File: solutions/0/0/1/0/p0010_s4.py
#!/usr/bin/env python3.14
# -*- coding: utf-8 -*-
""" Solution to Euler Problem 10: Summation of Primes [Level 0]. """
from __future__ import annotations

import primesieve
from solver.runners import runner


@runner.main
def solve(*args: str) -> str:
    """Sum primes below max_num via the primesieve library's optimised wheel-factorised sieve; ~O(n)."""
    max_num = runner.parse_int(args[0])

    return str(int(sum(primesieve.primes(max_num - 1))))


if __name__ == "__main__":
    raise SystemExit(solve())

File: solutions/public/p0010/p0010_s4.c
/* Solution to Euler Problem 10: Summation of Primes. */
#include "runner.h"
#include <primesieve.h>

/* Sum primes below max_num via the primesieve library's optimised wheel-factorised sieve; ~O(n). */
const char *solve(int argc, char *argv[]) {
    static char _answer[32];
    int max_num = parse_int(argv[1]);
    if (max_num < 2) { snprintf(_answer, sizeof _answer, "%lld", (long long)(0)); return _answer; }
    /* Sum of primes in [2, max_num - 1] */
    size_t size = 0;
    uint64_t *primes = (uint64_t *)primesieve_generate_primes(
        2, (uint64_t)(max_num - 1), &size, UINT64_PRIMES);
    if (!primes) { snprintf(_answer, sizeof _answer, "%lld", (long long)(0)); return _answer; }

    long long sum = 0;
    for (size_t i = 0; i < size; i++) {
        sum += (long long)primes[i];
    }
    primesieve_free(primes);
    { snprintf(_answer, sizeof _answer, "%lld", (long long)(sum)); return _answer; }
}
